### ISSUE TITLE
About the query transformation

### ISSUE THREAD

**Issue by soyoung97** [oriram/spider#10]
Hello, thanks for opening the amazing code!!
I'm actually trying to use this unsupervised pretraining method to passage-level WMT dataset,  (https://data.statmt.org/news-crawl/README) but it seems like there are some bugs (or maybe I did some mistake)

I have run the `preprocess_corpus.py`  with the `--compute_recurring_spans` option.
It seems like the code uses spacy to tokenize passages and find recurring spans, and then saves the passage id along with then index of the recurring spans based on the tokenizer(bert-base-uncased in my case) in `recurring_*.pkl`. (see the pic below) So, (20967, 164, 196) means (passage_id, span_start_idx, span_end_idx).
![image](https://user-images.githubusercontent.com/29880214/236675565-dc317ed1-5e2e-40cc-b4f2-5e0923617840.png)
full code looks something like this:
```
CUDA_VISIBLE_DEVICES=7 python preprocess_corpus.py \
--corpus_path path_to_passages.tsv \
--output_dir path_to_output_dir \
--tokenizer_name bert-base-uncased \
--num_processes 32  \
--compute_recurring_spans \
--min_span_length 2 \
--max_span_length 10
```
tsv file looks like this:
```
id      text    title
1       Wednesday, January 22, 2020. Boeing Pushes Back 737 Max Return Again The return of Boeing's 737 Max has been delayed again. On Tuesday, the company said it did not expect regulators to approve the jet to fly again until the middle of the year. American Airlines, United and Southwest had already taken Max flights off their schedules until June, and if this new timeline holds, it would further push back when the plane will be available for commercial flights. The announcement reflects Boeing's new appreciation for the challenges facing the company in its effort to persuade regulators that the Max is ready to fly. Boeing and the Federal Aviation Administration have continued to find new flaws with the Max that go beyond an automated software system known as MCAS. The software contributed to two accidents, in late 2018 and early 2019, that killed 346 people and led to the worldwide grounding of the Max in March. tsv_title_code_1
...
```

However, when running pre-training, about less than 1% of the data has incorrect matches of spans corresponding to the passages,
so the [original assertion](https://github.com/oriram/spider/blob/main/query_transformations.py#L147) sometimes doesn't work. Especially, there are some cases where span_**start_index** and span_**end_index** are **bigger than the passage length**. Please refer to the print output and the code I used to print it.
(I've also modified parts of the code to overcome edge cases. (for example, the original code didn't properly make queries for passage length that is smaller than min_window_size.))
![image](https://user-images.githubusercontent.com/29880214/236675817-5f1815b1-b13c-46d6-b9c3-2cb187968afd.png)
![image](https://user-images.githubusercontent.com/29880214/236675465-152251a9-b75f-4b4a-99c5-e334a1c7ed94.png)

In summary, I think there is something missing (or wrong) in assigning the index for preprocess_corpus.py. Do you have any idea on how to fix this..?

Thanks again for open-sourcing this code!!! It's helping me a lot on my work!

**Comment by oriram** [comment]
Hey @soyoung97,
Sorry for missing this issue :(
And, thanks for reaching out!

I guess it's not relevant anymore, but if so - did you set the maximal size for recurring spans? Also, are your passages of a constant length?

**Comment by soyoung97** [comment]
Hi, the passage length varies a bit, so we had to add some exception handling codes, but it seems to work fine now. Thanks for the reply!
